ant-protocol/src/storage/pointer.rs

```diff
@@ -96,7 +96,10 @@ impl Pointer {
     }
 
     /// Get the bytes that the signature is calculated from
-    fn bytes_to_sign(owner: &PublicKey, counter: u32, target: &PointerTarget) -> Vec<u8> {
+    fn bytes_to_sign(owner: &PublicKey, counter: u64, target: &PointerTarget) -> Vec<u8> {
+        // to support retrocompatibility with old pointers, we need to cast the counter to u32
+        let counter = counter as u32;
```

Comment:
There might be an attack vector here?

Imagine a pointer with counter `0` pointing to target `A`. Then an attacker can reset the pointer to target `A` by using a higher counter that casts to `0u32`.

Or does the counter update have to be `+1` compared to the previous?